Dockerfile

```diff
@@ -64,7 +64,12 @@ COPY --from=wasm-libs-builder /workspace/ /
 
 FROM wasm-base as wasm-bin-builder
     # pinned go version
-RUN curl -L https://golang.org/dl/go1.21.7.linux-`dpkg --print-architecture`.tar.gz | tar -C /usr/local -xzf -
+
+    # Use latest go1.21 version.
+RUN LATEST_GO=$(curl -s https://go.dev/dl/?mode=json | jq -r '[.[] | select(.version | startswith("go1.21"))] | .[0].version') && \
+    curl -L https://golang.org/dl/${LATEST_GO}.linux-`dpkg --print-architecture`.tar.gz | tar -C /usr/local -xzf - && \
+    echo "Installed Go version: ${LATEST_GO}"
```

Comment:
This will probably not help.
Original script already pulled latest version. The problem is once docekr caches it - it will not update.

Reply:
I think the original script was pinned to Go 1.21.7. But I think keeping it pinned makes sense. That way, we can at  least know what version of Go everyone's using, and we won't run into issues where one team turns out to be running a different Go version and that's the source of their issues.